### PR TITLE
Add Blog: Validating OpenAPI and JSON Schema

### PR DESCRIPTION
**Pull request by jdesrosiers**

### CODE REVIEW

**Comment by Relequestual** [comment]
Sooo looks like currently Cloudflare don't do deploy previews for PRs from a fork. *sigh*
So reviews will have to be local.

**Comment by Relequestual** [comment]
Do you want to pick an image or would you like me to suggest one? =D

**Review by Relequestual — CHANGES_REQUESTED** [review]
Can you add header information please?
Example: https://github.com/json-schema-org/blog/blob/dcc282d5b917e955a7f591d403b0488d270e6b81/pages/posts/code-of-conduct-for-json-schema.md?plain=1#L1-L14

**Comment by jdesrosiers** [comment]
I added header information. Personally, I think it's silly that every blog now has to have an image even if it has no relevance to the article. So I'll let you choose something. It doesn't matter to me.

**Comment by Relequestual** [comment]
> I added header information. Personally, I think it's silly that every blog now has to have an image even if it has no relevance to the article. So I'll let you choose something. It doesn't matter to me.

It doesn't have to, but it helps with drawing attention to the article when posting it on social media and in other places which show a preview of the content.

**Review by Relequestual — APPROVED** [review]

File: pages/posts/validating-openapi-and-json-schema.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    $ref: 'https://spec.openapis.org/oas/3.1/dialect/base'
+```
+
+That's enough to get the Schema Object validation we were after, but there are a
+few loose ends we'll want to tie up as well. The `jsonSchemaDialect` field in
+the OpenAPI 3.1 document can be used to change the dialect used. Since this
+schema only supports the default dialect, we want to restrict people from
+changing that to something else. If they need to change it, they'll need a
+different schema to validate against. We also don't want people using the
+`$schema` keyword to change the dialect of individual schemas.
```
Comment (karenetheridge) [inline pages/posts/validating-openapi-and-json-schema.md:176]: Is this intending to imply that it's prohibited to switch to a different metaschema by using the `$schema` keyword in an individual schema? I don't think that's correct -- [the docs for jsonSchemaDialect](https://spec.openapis.org/oas/v3.1.0#openapi-object) just say that this is the *default* value for the metaschema, not the only allowable value.

Comment (karenetheridge) [reply pages/posts/validating-openapi-and-json-schema.md:176]: (and indeed a subsequent example in this article goes on to show how to use a different metaschema for one particular schema.)

File: pages/posts/validating-openapi-and-json-schema.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+The `/components/schemas/foo` schema is understood to be interpreted as JSON
+Schema 2020-12 because that's what we set to be the default.
+
+The `/components/schemas/bar` schema changes the dialect of that schema to be
+JSON Schema draft-07. There are a couple of things working together to make this
+possible. The `$schema` keyword sets the dialect for the schema, but `$schema`
+is only allowed at the root of the document it appears in. That's why we also
+need to include the `$id` keyword. The `$id` keyword effectively makes that
+schema a separate document with it's own identifier and that location as the
```
Comment (karenetheridge) [inline pages/posts/validating-openapi-and-json-schema.md:248]: it's -> its

File: pages/posts/validating-openapi-and-json-schema.md (hunk 1 above)
Comment (Relequestual) [reply pages/posts/validating-openapi-and-json-schema.md:176]: I spent a little time thinking about this, and the intended implication is the context of "in this situation...". A hypothetical "we" which has imposed a specific governance rule for the work.

File: pages/posts/validating-openapi-and-json-schema.md (hunk 2 above)
Comment (Relequestual) [reply pages/posts/validating-openapi-and-json-schema.md:248]: Good catch. I'll make the change.

This PR was already merged, but this is still welcome. Thanks.
Feel free to make PRs if you see anything further.

Comment (Relequestual) [reply pages/posts/validating-openapi-and-json-schema.md:248]: https://github.com/json-schema-org/blog/commit/3ffd604bb77c3f5fea41213be4d46c24f7df820b

File: pages/posts/validating-openapi-and-json-schema.md (hunk 1 above)
Comment (karenetheridge) [reply pages/posts/validating-openapi-and-json-schema.md:176]: ah okay, fair enough.

**Pull request merged**